cloud/scope/cluster.go

```diff
@@ -108,11 +108,26 @@ func (s *ClusterScope) Subnets() infrav1.Subnets {
 	return s.AzureCluster.Spec.NetworkSpec.Subnets
 }
 
+// ControlPlaneSubnet returns the cluster control plane subnet.
+func (s *ClusterScope) ControlPlaneSubnet() *infrav1.SubnetSpec {
+	return s.AzureCluster.Spec.NetworkSpec.Subnets[0]
```

Comment:
I'd like to propose we add a `Role` field to the `SubnetSpec` so we can get the right subnet based on that instead of using the index in the list and requiring an order which isn't very resilient to human error. Adding the Role field would be breaking so I didn't want to do it as part of this PR.

Reply:
I support this proposal. It doesn't have to be breaking if we leverage mutating webhooks to set defaults.

Reply:
ok I can look into doing that